parachains/runtimes/assets/asset-hub-westend/src/xcm_config.rs

```diff
@@ -437,12 +441,15 @@ impl xcm_executor::Config for XcmConfig {
 	>;
 	type Trader = (
 		UsingComponents<WeightToFee, WestendLocation, AccountId, Balances, ToStakingPot<Runtime>>,
-		cumulus_primitives_utility::TakeFirstAssetTrader<
-			AccountId,
-			AssetFeeAsExistentialDepositMultiplierFeeCharger,
-			TrustBackedAssetsConvertedConcreteId,
-			Assets,
+		cumulus_primitives_utility::SwapFirstAssetTrader<
```

Comment:
@PatricioNapoli 

I wouldnt use `LocalAndForeignAssets` here,

what about having multiple `SwapFirstAssetTrader` per pallet_asset instance?
this could pretty simplify it and all matcher are ready and already in use, like:

```
// swap for Local Assets
cumulus_primitives_utility::SwapFirstAssetTrader<
			Runtime,
			LocationToAccountId,
			pallet_asset_conversion::Pallet<Runtime>,
			WeightToFee,
			TrustBackedAssetsConvertedConcreteId,
			Assets,
            ...
>,
// swap for Foreign Assets
cumulus_primitives_utility::SwapFirstAssetTrader<
			Runtime,
			LocationToAccountId,
			pallet_asset_conversion::Pallet<Runtime>,
			WeightToFee,
			ForeignAssetsConvertedConcreteId,
            ForeignAssets,
            ...
>,
```